agent/eventhandler/task_handler.go

```diff
@@ -58,16 +67,73 @@ type TaskHandler struct {
 	// stateSaver is a statemanager which may be used to save any
 	// changes to a task or container's SentStatus
 	stateSaver statemanager.Saver
+
+	drainEventsFrequency time.Duration
+	state                dockerstate.TaskEngineState
+	client               api.ECSClient
+	ctx                  context.Context
 }
 
 // NewTaskHandler returns a pointer to TaskHandler
-func NewTaskHandler(stateManager statemanager.Saver) *TaskHandler {
-	return &TaskHandler{
+func NewTaskHandler(ctx context.Context,
+	stateManager statemanager.Saver,
+	state dockerstate.TaskEngineState,
+	client api.ECSClient) *TaskHandler {
+	// Create a handler and start the periodic event drain loop
+	taskHandler := &TaskHandler{
+		ctx:                    ctx,
 		tasksToEvents:          make(map[string]*eventList),
 		submitSemaphore:        utils.NewSemaphore(concurrentEventCalls),
 		tasksToContainerStates: make(map[string][]api.ContainerStateChange),
 		stateSaver:             stateManager,
+		state:                  state,
+		client:                 client,
+		drainEventsFrequency:   drainEventsFrequency,
+	}
+	go taskHandler.startDrainEventsTicker()
+
+	return taskHandler
+}
+
+// startDrainEventsTicker starts a ticker that periodically drains the events queue
+// by submitting state change events to the ECS backend
+func (handler *TaskHandler) startDrainEventsTicker() {
+	ticker := time.NewTicker(handler.drainEventsFrequency)
+	for {
+		select {
+		case <-handler.ctx.Done():
+			seelog.Infof("TaskHandler: Stopping periodic container state change submission ticker")
+			ticker.Stop()
+			return
+		case <-ticker.C:
+			seelog.Info("TaskHandler tick")
+			for _, event := range handler.getBatchedContainerEvents() {
+				seelog.Infof(
```

Comment:
Same question as above. Should this be `Debug`?

Reply:
it makes sense for this to be `Info` as it's a trigger for submitting state changes to the backend and we'd want to know about it without having to elevate log level to debug